app/models/user_mailbox.rb

```diff
@@ -0,0 +1,44 @@
+class UserMailbox
+  attr_reader :user
+
+  def initialize(user)
+    raise ArgumentError.new("Must provide a user") if user == nil
```

Comment:
Not catching this in the codebase raises my hackles too, because that means if it triggers, users will see an error message (and one that means nothing to them).

Under what conditions will an instance of this class get created with a nil user? What situation are we trying to prevent?